pkg/controller/webhook/admission_registration.go

```diff
@@ -0,0 +1,162 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package webhook
+
+import (
+	"context"
+
+	v1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type webhook struct {
+	webhookConfigurationName, webhookName string
+	caBundle                              []byte
+}
+
+type Services map[types.NamespacedName]struct{}
+
+// AdmissionControllerInterface helps to setup webhooks for different versions of the admissionregistration API.
+type AdmissionControllerInterface interface {
+	getType() client.Object
+	// services returns the set of services used by the Webhooks
+	services() Services
+	// webhooks returns the list of webhook in the configuration
+	webhooks() []webhook
+	// update ca bundle with the provided CA in all the Webhooks
```

Comment:
```suggestion
	// update CABundle with the provided CA in all the Webhooks
```
not sure. or "CA bundle" maybe